docs/reference/tools-reference/open_source_llm_tool.md

```diff
@@ -0,0 +1,59 @@
+# Open Source LLM
+
+## Introduction
+
+The Prompt flow Open Source LLM tool enables you to utilize a variety of Open Source and Foundational Models, such as [Falcon](https://aka.ms/AAlc25c) or [Llama 2](https://aka.ms/AAlc258) for natural language processing, in PromptFlow.
+
+This Prompt flow supports two different LLM API types:
+
+- **Completion**: to generate text based on provided prompts.
+- **Chat**: to facilitate interactive conversations with text-based inputs and responses.
+
+## Prerequisite
+
+1. Pick the model to use with from the [Azure Machine Learning model catalog](https://ml.azure.com/model/catalog).
+2. Use the "Deploy" button to deploy the model to AzureML Inference endpoint.
+
+More detailed instructions can be found here [Deploying foundation models to endpoints for inferencing.](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-use-foundation-models?view=azureml-api-2#deploying-foundation-models-to-endpoints-for-inferencing)
+
+## **Connections**
+
+Setup connections to provisioned resources in prompt flow.
+
+| Type        | Name     | API KEY  | API Type | API Version |
+|-------------|----------|----------|----------|-------------|
+| CustomConnection | Required | Required | -        | -           |
+
+Instructions to create a Custom Connection [can be found here.](https://learn.microsoft.com/en-us/azure/machine-learning/prompt-flow/how-to-integrate-with-langchain?view=azureml-api-2#create-a-connection)
+
+The keys to set are:
+
+1. endpoint_url
+2. endpoint_api_key
```

Comment:
The `endpoint_api_key` should be set in secrets. If set it in keys, will get this error 
![image](https://github.com/microsoft/promptflow/assets/95729303/4b6b329e-8bbd-4c3c-b8b4-bf411439d1e2)